l10n_be_cooperator/models/tax_shelter_declaration.py

```diff
@@ -155,7 +155,9 @@ def compute_declaration(self):
 
     def validate_declaration(self):
         self.ensure_one()
-        self.tax_shelter_certificates.write({"state": "validated"})
+        for certificate in self.tax_shelter_certificates:
+            if certificate.state == "draft":
+                certificate.write({"state": "validated"})
```

Comment:
```suggestion
        self.tax_shelter_certificates.filtered(lambda x: x.state == "draft").write(
            {"state": "validated"}
        )
```

Reply:
done ! does this improve performances ?

Reply:
i’m not sure how odoo does it, but like this it might do the operation in only one sql update, while with a for loop it would be forced to do an update per record.